src/core/lib/channel/connected_channel.cc

```diff
@@ -824,11 +824,19 @@ ArenaPromise<ServerMetadataHandle> MakeServerCallPromise(
   auto cleanup_polling_entity_latch =
       std::unique_ptr<Latch<grpc_polling_entity>, CleanupPollingEntityLatch>(
           &call_data->polling_entity_latch);
+  struct CleanupSendInitialMetadata {
+    void operator()(CallData* call_data) {
+      call_data->server_initial_metadata.receiver.CloseWithError();
+    }
+  };
+  auto cleanup_send_initial_metadata =
```

Comment:
The `auto` doesn't seem to add much here.  Seems like this can be:

```
std::unique_ptr<CallData, CleanupSendInitialMetadata> cleanup_send_initial_metadata(
    call_data);
```

Alternatively, maybe use `absl::Cleanup` here?